src/components/ContentWrapper.jsx

```diff
@@ -28,6 +44,42 @@ const ContentWrapper = ({ hasServiceBeenLaunched }) => {
   )
 }
 
+const ContentWrapper = ({ hasServiceBeenLaunched }) => {
+  return (
+    <Query
+      query={contactsByFamilyNameGivenNameEmailCozyUrl.definition}
+      as={contactsByFamilyNameGivenNameEmailCozyUrl.options.as}
+    >
+      {contactsWithIndexesResult => {
+        if (hasQueryMoreToFetch(contactsWithIndexesResult)) {
+          contactsWithIndexesResult.fetchMore()
+        }
+
+        return (
+          <Query
+            query={contactsWithoutIndexes.definition}
+            as={contactsWithoutIndexes.options.as}
+          >
+            {contactsWithNoIndexesResult => {
+              if (hasQueryMoreToFetch(contactsWithNoIndexesResult)) {
+                contactsWithNoIndexesResult.fetchMore()
+              }
+
+              return (
+                <ContentWrapperResult
+                  hasServiceBeenLaunched={hasServiceBeenLaunched}
+                  contactsWithIndexesResult={contactsWithIndexesResult}
+                  contactsWithNoIndexesResult={contactsWithNoIndexesResult}
+                />
+              )
+            }}
+          </Query>
+        )
+      }}
+    </Query>
+  )
+}
```

Comment:
There is already a spinner until both are finished. It's managed by `ContentWrapperResult`.

Yes we can show a spinner while the first one is not finished, and only then launch the second query. This will make no performance difference, but a visual one : when the first one is finished, there is again a spinner for the second one, with a little visual glitch (because the second one start at default position, not where the first one was before)

Reply:
> This will make no performance difference 

Why that? If the first query changes (let's say its fetchStatus), it will rerender its child, and then re-render que second query (and then make again the http call since there is no fetchPolicy on it). 

Here, if the first query changes and so re-render, all the children

Reply:
Oh right, Query makes the call in DidMount. https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/Query.jsx#L116

Reply:
There is 2 POST by queries : one for `io.cozy.contacts/_index` and one for `io.cozy.contacts/_find`

I ran a test and there is the results :

- With the actual code :
query1 : 3 renders
query2 : 6 renders
total of 3 couples of POST

Behavior :
render query1 then query2 then
trigger 2 couples of POST (1 initial one by queries) then
render query2 then query1 then query2 then
trigger 1 couple of POST (query1 fetchMore) then
render query2 then query2 then query1 then query2

- With spinner while query1 has not finished to fetch all datas
query1 : 3 renders
query2 : 2 renders
total of 3 couples of POST

Behavior :
render query1 then
trigger 1 couple of POST (initial one) then
render query1 then
trigger 1 couple of POST (query1 fetchMore) then
render query1 then query2 then
trigger 1 couple of POST (query2 initial one) then
render query2

Reply:
I just tested it with 6400+ contacts and there is no performance difference when the list shows up.

Reply:
What do you think ?